packages/ai-core/src/browser/frontend-prompt-customization-service.ts

```diff
@@ -486,59 +519,100 @@ export class DefaultPromptFragmentCustomizationService implements PromptFragment
         }
         this.onDidChangePromptFragmentCustomizationEmitter.fire(Array.from(parsedPromptFragments));
         this.onDidChangeCustomAgentsEmitter.fire();
+    }
 
+    /**
+     * Sets up file watching for a template directory (works for both existing and non-existing directories)
+     * @param dirURI URI of the directory to watch
+     * @param priority Priority level for customizations in this directory
+     * @param customizationSource Source type of the customization
+     */
+    protected setupDirectoryWatcher(
+        dirURI: URI,
+        priority: number,
+        customizationSource: CustomizationSource
+    ): void {
         this.toDispose.push(this.fileService.watch(dirURI, { recursive: true, excludes: [] }));
         this.toDispose.push(this.fileService.onDidFilesChange(async (event: FileChangesEvent) => {
-            // Only watch for changes within provided dir
+            // Filter for changes within the watched directory
             if (!event.changes.some(change => change.resource.toString().startsWith(dirURI.toString()))) {
                 return;
             }
+
+            // Handle directory creation (when watching a previously non-existent directory)
+            if (event.getAdded().some(addedFile => addedFile.resource.toString() === dirURI.toString())) {
```

Comment:
This works for the case that a template directory was added. However we should also update when a template directory was deleted.